Submitty/Submitty
# [Refactor:Router] New URL for ForumController

### URL Changes

- POST `http://192.168.56.111/f19/sample/forum/search`
- POST `http://192.168.56.111/f19/sample/forum/announcements`
- GET `http://192.168.56.111/f19/sample/forum/stats`
- GET `http://192.168.56.111/f19/sample/forum/categories`
- POST `http://192.168.56.111/f19/sample/forum/categories/new`
- POST `http://192.168.56.111/f19/sample/forum/categories/delete`
- POST `http://192.168.56.111/f19/sample/forum/categories/edit`
- POST `http://192.168.56.111/f19/sample/forum/categories/reorder`
- GET or POST `http://192.168.56.111/f19/sample/forum/threads`
- GET or POST `http://192.168.56.111/f19/sample/forum/threads/{thread_id}`
- GET or POST `http://192.168.56.111/f19/sample/forum/threads/new`
- POST `http://192.168.56.111/f19/sample/forum/threads/status`
- POST `http://192.168.56.111/f19/sample/forum/threads/pin`
- POST `http://192.168.56.111/f19/sample/forum/threads/merge`
- POST `http://192.168.56.111/f19/sample/forum/posts/new`
- POST `http://192.168.56.111/f19/sample/forum/posts/history`
- POST `http://192.168.56.111/f19/sample/forum/posts/get`

### Testing
Manually tested the following:
- searching
- make announcements
- see stats
- see categories, adding, deleting, reordering.
- resolve & unresolve threads
- pin & unpin threads
- merge threads
- make & edit posts
- notification

### Other Info
I feel `POST` is overused in ForumController... Due to the extensive nature of this controller and Anu working on this, I didn't modify much inside the function. I don't think many of them should be `POST` though.
